internal/internal_worker.go

```diff
@@ -1083,18 +1055,24 @@ func getBinaryChecksum() string {
 
 // Run the worker in a blocking fashion. Stop the worker when process is killed with SIGINT or SIGTERM.
 // Returns error only if worker fails to start.
-func (aw *AggregatedWorker) Run() error {
+func (aw *AggregatedWorker) Run(ch <-chan interface{}) error {
 	if err := aw.Start(); err != nil {
 		return err
 	}
-	d := <-getKillSignal()
-	aw.logger.Info("Worker has been killed", zap.String("Signal", d.String()))
-	aw.Stop()
+	select {
+	case s := <-ch:
+		aw.logger.Info("Worker has been killed", zap.String("Signal", fmt.Sprintf("%v", s)))
```

Comment:
No 100% sure if this message is helpful if user controlled channel is passed as an argument.

Reply:
Well, it is still signal. Not `os.Signal` but "signal to stop worker". Rephrased message though.